src/js/pages/Search/index.js

```diff
@@ -20,8 +21,58 @@ export default class Search extends React.PureComponent {
     writer: PropTypes.string,
   };
 
+  state = {
+    searchURL: '',
+    verseIdList: [],
+    semanticInfo: {}
+  };
+
+  setSearchUrl() {
+    const { q, type, source, offset, writer } = this.props;
+    const isChatBot = type === SEARCH_TYPES.ASK_A_QUESTION;
+    let semanticInfo;
+
+    if (isChatBot) {
+      const semanticApi = encodeURI(`${GURBANIBOT_URL}search/?query=${q}`);
+      const semanticReq = fetch(semanticApi).then((response) => response.json());
+      semanticReq.then((semanticData) => {
+        const shabadIdList = semanticData.results.map((dataObj) => {
+          const { ShabadID, VerseID } = dataObj.Payload;
+          this.state.verseIdList.push(VerseID);
```

Comment:
Can we push to some other array here and update the state just once after the map?

Reply:
Done, thanks ji!